packages/framework/aqueduct/src/data-objects/dataObject.ts

```diff
@@ -40,10 +41,20 @@ export abstract class DataObject<O extends IFluidObject = object, S = undefined,
 
     public async request(request: IRequest): Promise<IResponse> {
         const url = request.url;
-        if (url.startsWith(this.bigBlobs)) {
-            const value = this.root.get<string>(url);
+        const requestParser = RequestParser.create({ url: request.url });
+        const itemId = requestParser.pathParts[0];
+        if (itemId === "bigBlobs") {
+            let value = this.root.get<string>(url);
             if (value === undefined) {
-                return { mimeType: "text/plain", status: 404, value: `request ${url} not found` };
+                const value2 = this.root.get<string>(url.slice(1));
+                if (value2 === undefined) {
+                    return { mimeType: "text/plain", status: 404, value: `request ${url} not found` };
+                }
+                else {
+                    this.root.set(url.slice(1),value2);
+                    this.root.delete(url);
```

Comment:
I know I suggested this change, but after more thought I don't think there's value in converting the format here -- I don't think there's really a benefit from having the key format exactly match the URL (even though it currently does today).

This way you can just join the path parts with '/' to get the key, and thereby avoid the slice calls, something like

```typescript
if (requestParser.pathParts[0] === this.bigBlobs) {
    const value = this.root.get<string>(requestParser.pathParts.join("/"));
    // ... the rest
}
```